### ISSUE TITLE
[Feat] Add Streaming for O-1 models 

### ISSUE THREAD

**Issue by m-hoseyny** [BerriAI/litellm#5694]
### What happened?

Hello

I am using litellm 1.45.0 as proxy server. I've go an error when I am calling o1 family. The error is:

```
'async for' requires an object with __aiter__ method, got ModelResponse

Traceback (most recent call last):
File "/usr/local/lib/python3.11/site-packages/litellm/proxy/proxy_server.py", line 2586, in async_data_generator
async for chunk in response:
TypeError: 'async for' requires an object with __aiter__ method, got ModelResponse
```

I checked the code and it seems that the openwebui send the request as `stream=true` and the litellm is returning the result with async streaming function. Is it possible to solve it or add a config like `support_stream` to disable streaming for o1 family?

**Comment by ghost** [comment]
Hi @m-hoseyny faking o-1 streaming support should be done today. Here's the parent issue to track this - https://github.com/BerriAI/litellm/issues/5694

**Comment by ishaan-jaff** [comment]
The o-1 family does not support streaming so litellm needs to fake o-1 streaming

**Comment by ishaan-jaff** [comment]
tracking here @m-hoseyny https://github.com/BerriAI/litellm/issues/5672

**Comment by m-hoseyny** [comment]
Thanks for follow-up.
I updated the system to version 1.46 and I saw you add the stream and stream_option. I am using openwebui and it sends the requests by `stream=True` in the header you are directly sent the params to openai and it raises error about stream=false in o1 family. How can i configure the litellm to change the param to send the request to openai and change the params?
Note: I add the params in the `litellm_params` in the database but i did not get any response
